### PR TITLE
Bounty Creation Unit Test

### PR DESCRIPTION
**Pull request by Khaalid-oh**
## Issue ticket number and link

Closes https://github.com/stakwork/sphinx-tribes/issues/996

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist before requesting a review
- [ ] I have performed a self-review of my code
Screenshot 2023-12-02 at 01 39 01

- [ ] I have tested on Chrome and Firefox
- [ ] I have tested on a mobile device
- [ ] I have provided a screenshot or recording of changes in my PR if there were updates to the frontend
- [ ] Screenshot 2023-12-02 at 01 39 01

### CODE REVIEW

**Comment by kevkevinpal** [comment]
looks like gitactions checks are failing if you can fix

**Comment by ecurrencyhodler** [comment]
Hey @Khaalid-oh thanks for this PR. It looks like some tests are failing such as prettier, craco, and jest. Can you address those? If you're not familiar with how, you can join our stand up and you cask raph/kev for help.

**Comment by Khaalid-oh** [comment]
> Hey @Khaalid-oh thanks for this PR. It looks like some tests are failing such as prettier, craco, and jest. Can you address those? If you're not familiar with how, you can join our stand up and you cask raph/kev for help.

Alright, I'm sure he can help with some of the questions I have. You can set that up

**Comment by ecurrencyhodler** [comment]
Invited

**Comment by kevkevinpal** [comment]
made some updates in [d786697](https://github.com/stakwork/sphinx-tribes/pull/1037/commits/d786697e174f4568a55291535711fabbe08254dd)

I think we still need to add some changes but I got started a bit but we still probably need to make some changes let me know what you think @Khaalid-oh

**Comment by Khaalid-oh** [comment]
> made some updates in [0293310](https://github.com/stakwork/sphinx-tribes/pull/1037/commits/0293310b6ee5a9b45788b4aec275cc2a5313eaed)
> 
> I think we still need to add some changes but I got started a bit but we still probably need to make some changes let me know what you think @Khaalid-oh

Looks nice to me, what other changes do you have in mind?

**Comment by ecurrencyhodler** [comment]
Hey so we are missing two tests from the issue:

- [ ] Get people bounty test
- [ ] Delete bounty

**Comment by Khaalid-oh** [comment]
> Hey so we are missing two tests from the issue:
> 
> * [ ]  Get people bounty test
> * [ ]  Delete bounty

Alright, I'm on it

**Comment by ecurrencyhodler** [comment]
All tests passed. Let's get one more code review

File: frontend/app/src/config/host.ts [hunk 1]
```diff
@@ -3,7 +3,7 @@ const externalDockerHosts = ['localhost:23007', 'localhost:23000'];
 
 export function getHost(): string {
   const host = window.location.host.includes('localhost') ? 'localhost:5002' : window.location.host;
-  return host;
+  return 'people.sphinx.chat';
```
Comment (kevkevinpal) [inline frontend/app/src/config/host.ts:6]: please change this back, this will also effect the unit tests

File: frontend/app/src/pages/people/PersonPage.tsx
Comment (kevkevinpal) [inline frontend/app/src/pages/people/PersonPage.tsx:1]: not sure why there is code change for a unit test PR

File: frontend/app/yarn.lock
Comment (kevkevinpal) [inline frontend/app/yarn.lock:1]: we shouldnt be including a new yarn.lock file

File: frontend/app/src/store/main.ts [hunk 2]
```diff
@@ -1428,6 +1428,7 @@ export class MainStore {
 
   async deleteBounty(created: number, owner_pubkey: string): Promise<void> {
     const info = uiStore.meInfo as any;
+    console.log(uiStore.meInfo);
```
Comment (kevkevinpal) [inline frontend/app/src/store/main.ts:1431]: remove console.log I think I may have added this on accident

File: frontend/app/src/people/widgetViews/organization/AddUserModal.tsx [hunk 3]
```diff
@@ -62,7 +62,7 @@ const AddUserModal = (props: AddUserModalProps) => {
         ...(config?.modalStyle ?? {}),
         maxHeight: '100%',
         borderRadius: '10px',
-        minWidth: '22rem'
+        maxWidth: '22rem'
```
Comment (kevkevinpal) [inline frontend/app/src/people/widgetViews/organization/AddUserModal.tsx:65]: not sure why this is needed on a unit test PR

File: frontend/app/src/people/userInfo/UserInfoDesktopView.tsx
Comment (kevkevinpal) [inline frontend/app/src/people/userInfo/UserInfoDesktopView.tsx:1]: Revert this, there should be no changes on a unit test PR

File: frontend/app/src/bounties/__tests__/bountyTest.spec.ts [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+//           twitter_confirmed: true,
+//           unique_name: 'umbreltest',
+//           unlisted: false,
+//           updated: '2023-12-04T19:48:05.641056Z',
+//           uuid: 'cd9dm5ua5fdtsj2c2nbg',
+//           wanteds: []
+//         }
+//       }
+//     ]),
+// }));
```
Comment (kevkevinpal) [inline frontend/app/src/bounties/__tests__/bountyTest.spec.ts:73]: we should delete this comment

File: frontend/app/src/bounties/__tests__/bountyTest.spec.ts [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+
+const mockedApi = api as jest.Mocked<typeof api>;
+//const mockedUiStore = uiStore as jest.Mocked<typeof uiStore>;
```
Comment (kevkevinpal) [inline frontend/app/src/bounties/__tests__/bountyTest.spec.ts:76]: delete comment

File: frontend/app/src/bounties/__tests__/bountyTest.spec.ts [hunk 6]
```diff
@@ -0,0 +1,238 @@
...
+
+describe('Bounty Tests', () => {
+  beforeAll(() => {
+    Object.defineProperty(window, 'localStorage', { value: mockLocalStorage });
+  });
+
+  beforeEach(() => {
+    window.localStorage.clear();
+
+    //window.localStorage.setItem('peopleBounties', JSON.stringify(mockBounties));
```
Comment (kevkevinpal) [inline frontend/app/src/bounties/__tests__/bountyTest.spec.ts:86]: delete comment

File: frontend/app/src/bounties/__tests__/bountyTest.spec.ts [hunk 7]
```diff
@@ -0,0 +1,238 @@
...
+  });
+
+  afterEach(() => {
+    window.localStorage.clear();
+  });
+
+  //it('should save and retrieve bounties from local storage', () => {
+  //  const storedBounties = JSON.parse(window.localStorage.getItem('peopleBounties') || '[]');
+  //  expect(storedBounties).toEqual(mockBounties);
+  //}, 8000);
```
Comment (kevkevinpal) [inline frontend/app/src/bounties/__tests__/bountyTest.spec.ts:96]: delete comment

File: frontend/app/src/bounties/__tests__/bountyTest.spec.ts [hunk 8]
```diff
@@ -0,0 +1,238 @@
...
+
+  it('should fetch bounties and save to localStorage', async () => {
+    mockedApi.get = jest.fn().mockResolvedValue(mockBounties);
+    const bounties = await mainStore.getPeopleBounties({ resetPage: true });
+    expect(mockedApi.get).toHaveBeenCalledTimes(1);
+    expect(bounties).toStrictEqual(mockBountiesMutated);
+  });
+
+  it('should save a new bounty and persist to localStorage', async () => {
+    //mockedApi.post.mockResolvedValue({ data: newBounty });
```
Comment (kevkevinpal) [inline frontend/app/src/bounties/__tests__/bountyTest.spec.ts:106]: delete comment

File: frontend/app/src/bounties/__tests__/bountyTest.spec.ts [hunk 9]
```diff
@@ -0,0 +1,238 @@
...
+        body: JSON.stringify(newBounty),
+        headers: { 'Content-Type': 'application/json', 'x-jwt': undefined },
+        method: 'POST',
+        mode: 'cors'
+      }
+    ];
+
+    expect(global.fetch).toHaveBeenCalledWith(...postRequestContent);
+    //const bounties = JSON.parse(mockLocalStorage.getItem('peopleBounties'));
+    //expect(window.localStorage.getItem('peopleBounties')).toContainEqual([]);
```
Comment (kevkevinpal) [inline frontend/app/src/bounties/__tests__/bountyTest.spec.ts:125]: delete comment

File: frontend/app/src/bounties/__tests__/bountyTest.spec.ts [hunk 10]
```diff
@@ -0,0 +1,238 @@
...
+          tags: [],
+          twitter_confirmed: true,
+          unique_name: 'umbreltest',
+          unlisted: false,
+          updated: '2023-12-04T19:48:05.641056Z',
+          uuid: 'cd9dm5ua5fdtsj2c2nbg',
+          wanteds: []
+        }
+      }
+    ];
```
Comment (kevkevinpal) [inline frontend/app/src/bounties/__tests__/bountyTest.spec.ts:192]: we should move this into the `mockBounties.data.ts` file

File: frontend/app/src/people/userInfo/UserInfoDesktopView.tsx
Comment (Khaalid-oh) [reply frontend/app/src/people/userInfo/UserInfoDesktopView.tsx:1]: This is for another issue raised, I'll create a new PR for this

File: frontend/app/src/config/host.ts (hunk 1 above)
Comment (Khaalid-oh) [reply frontend/app/src/config/host.ts:6]: This is also for another issue, I'll create a different PR for it

File: frontend/app/src/people/widgetViews/organization/AddUserModal.tsx (hunk 3 above)
Comment (kevkevinpal) [reply frontend/app/src/people/widgetViews/organization/AddUserModal.tsx:65]: not sure why you marked this as resolved

File: frontend/app/yarn.lock
Comment (kevkevinpal) [inline frontend/app/yarn.lock:1]: remove the yarn.lock since no new packages were added

File: frontend/app/src/people/userInfo/UserInfoDesktopView.tsx
Comment (kevkevinpal) [inline frontend/app/src/people/userInfo/UserInfoDesktopView.tsx:1]: why is this file modified

File: frontend/app/src/pages/people/PersonPage.tsx
Comment (kevkevinpal) [inline frontend/app/src/pages/people/PersonPage.tsx:1]: why is this file modifed

Comment (Khaalid-oh) [reply frontend/app/src/pages/people/PersonPage.tsx:1]: This is for the  [glitch when moving from a tab on profile to default profile view]

Comment (kevkevinpal) [reply frontend/app/src/pages/people/PersonPage.tsx:1]: it should not be modified in this PR
